test-manager/src/tests/mod.rs

```diff
@@ -115,1259 +119,1337 @@ macro_rules! get_possible_api_endpoints {
+) -> Option<&'static Settings> {
+    match DEFAULT_SETTINGS.get() {
+        None => {
+            let settings: Settings = mullvad_client
+                .get_settings(())
+                .await
+                .map_err(|_error| Error::DaemonError(String::from("Could not get settings")))
+                .ok()?
+                .into_inner();
+            DEFAULT_SETTINGS.set(settings).unwrap();
+            DEFAULT_SETTINGS.get()
         }
+        Some(settings) => Some(settings),
+    }
+}
 
-        // install package
-        let mut ctx = context::current();
-        ctx.deadline = SystemTime::now().checked_add(INSTALL_TIMEOUT).unwrap();
+/// Takes Optional default settings and Optional management interfaces of both new and old types
+/// If no default settings or neither management interface is provided then does no daemon related
+/// cleanup
+pub async fn cleanup_after_test(
+    default_settings: Option<&Settings>,
+    mullvad_client: Option<ManagementServiceClient>,
+) -> Result<(), Error> {
+    match mullvad_client {
+        Some(mut mullvad_client) => {
+            log::debug!("Cleaning up daemon in test cleanup");
+            if let Some(default_settings) = default_settings {
+                mullvad_client
+                    .set_allow_lan(default_settings.allow_lan)
+                    .await
+                    .expect("Could not set allow lan in cleanup");
+                mullvad_client
+                    .set_show_beta_releases(default_settings.show_beta_releases)
+                    .await
+                    .expect("Could not set show beta releases in cleanup");
+                mullvad_client
+                    .set_bridge_settings(default_settings.bridge_settings.clone().unwrap())
+                    .await
+                    .expect("Could not set bridge settings in cleanup");
+                mullvad_client
+                    .set_obfuscation_settings(default_settings.obfuscation_settings.clone().unwrap())
+                    .await
+                    .expect("Could set obfuscation settings in cleanup");
+                mullvad_client
+                    .set_block_when_disconnected(default_settings.block_when_disconnected)
+                    .await
+                    .expect("Could not set block when disconnected setting in cleanup");
+                mullvad_client
+                    .clear_split_tunnel_apps(())
+                    .await
+                    .expect("Could not clear split tunnel apps in cleanup");
+                mullvad_client
+                    .clear_split_tunnel_processes(())
+                    .await
+                    .expect("Could not clear split tunnel processes in cleanup");
+            }
 
-        rpc.install_app(ctx, get_package_desc(&rpc, &*PREVIOUS_APP_FILENAME).await?)
-            .await?
-            .map_err(|err| Error::Package("previous app", err))?;
+            reset_relay_settings(&mut mullvad_client).await?;
```

Comment:
There might be a way to use `settings` for these as well. Also, we could get rid of all `reset_relay_settings` calls elsewhere.

Reply:
I will look at this